ts/store/reducers/crossSessions.ts

```diff
@@ -0,0 +1,59 @@
+/**
+ * this state / reducer represents and handles all those data should be kept across multiple users sessions
+ */
+import { FiscalCode, NonEmptyString } from "italia-ts-commons/lib/strings";
+import { isActionOf } from "typesafe-actions";
+import { createSelector } from "reselect";
+import { fromNullable } from "fp-ts/lib/Option";
+import sha from "sha.js";
+import { Action } from "../actions/types";
+import { setProfileHashedFiscalCode } from "../actions/crossSessions";
+import { GlobalState } from "./types";
+
+type HashedFiscalCode = NonEmptyString | undefined;
+export type CrossSessionsState = {
+  hashedFiscalCode: HashedFiscalCode;
+};
+
+const INITIAL_STATE: CrossSessionsState = {
+  hashedFiscalCode: undefined
+};
+
+const hash = (value: string): string =>
+  sha("sha256").update(value).digest("hex");
+
+const reducer = (
+  state: CrossSessionsState = INITIAL_STATE,
+  action: Action
+): CrossSessionsState => {
+  if (isActionOf(setProfileHashedFiscalCode, action)) {
+    return {
+      ...state,
+      hashedFiscalCode: hash(action.payload) as NonEmptyString
```

Comment:
Maybe you can avoid `as NonEmptyString` modifying, the response type of the `hash` function?
```
const hash = (value: string): HashedFiscalCode =>
  NonEmptyString.decode(sha("sha256").update(value).digest("hex")).fold(
    _ => undefined,
    hFC => hFC
  );
```

Reply:
I refactored the code 
https://github.com/pagopa/io-app/pull/2732/commits/75147ba767c432ef32cc1a6534caa4600fb97627